File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> result;
        vector<int> curr;
        generateSubsets(0, nums, curr, result);
        return result;
    }

private:
    void generateSubsets(int index, const vector<int>& nums, vector<int>& curr, vector<vector<int>>& result) {
        if (index == nums.size()) {
            result.push_back(curr);
            return;
        }
        
        // Exclude current element
        generateSubsets(index + 1, nums, curr, result);
        
        // Include current element
        curr.push_back(nums[index]);
        generateSubsets(index + 1, nums, curr, result);
        curr.pop_back(); // Backtrack
    }
};

File: 0078-subsets/0078-subsets.py

class Solution(object):
    def __init__(self):
        self.ans = []

    def subsets(self, nums):
        def sub(i, temp):
            if i == len(nums):
                self.ans.append(temp[:])  # Append a copy of 'temp' to the result
                return

            sub(i + 1, temp)  # Exclude the current element
            temp.append(nums[i])  # Small work: Include the current element
            sub(i + 1, temp)
            temp.pop()  # Backtrack by removing the current element

        temp = []
        sub(0, temp)
        return self.ans      